### PR TITLE
Create Fork Tests for Add/Remove Liquidity from Buffer

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Description

Create fork tests to make sure add and remove liquidity from buffer are not profitable to the user.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol [hunk 1]
```diff
@@ -197,20 +212,90 @@ abstract contract ERC4626WrapperBaseTest is Test {
...
 
-    function _initializeUserWallet() private {
-        (user, ) = makeAddrAndKey("User");
-        vm.label(user, "User");
+    function testAddLiquidityToBuffer__Fork__Fuzz(
+        uint256 underlyingToInitialize,
+        uint256 wrappedToInitialize,
+        uint256 sharesToIssue
+    ) public {
+        underlyingToInitialize = bound(underlyingToInitialize, 1e6, underlyingToken.balanceOf(lp) / 10);
```
Comment (EndymionJkb) [inline pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol:220]: Are these minimum numbers deliberate? They don't seem to correspond to any of the relevant constants (e.g., MIN_WRAP_AMOUNT, _BUFFER_MINIMUM_TOTAL_SUPPLY)

File: pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol [hunk 2]
```diff
@@ -197,20 +212,90 @@ abstract contract ERC4626WrapperBaseTest is Test {
...
+        // value returned by initializeBuffer;
+        uint256 totalShares = vault.getBufferTotalShares(wrapper);
+
+        sharesToRemove = bound(sharesToRemove, 0, lpShares);
+
+        vm.prank(lp);
+        (uint256 underlyingRemoved, uint256 wrappedRemoved) = vault.removeLiquidityFromBuffer(wrapper, sharesToRemove);
+
+        // Measures if the underlying and wrapped received from `removeLiquidityFromBuffer` are worth the same number
+        // of shares than in initialized liquidity (or less).
```
Comment (EndymionJkb) [inline pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol:267]:
```suggestion
        // Ensure the underlying and wrapped value returned from `removeLiquidityFromBuffer` is equal to or less than
        // the amount received from initialization.
```

File: pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol [hunk 3]
```diff
@@ -197,20 +212,90 @@ abstract contract ERC4626WrapperBaseTest is Test {
...
+        router.initializeBuffer(wrapper, underlyingToInitialize, wrappedToInitialize);
+        // Since the buffer burns part of the shares, we measure the total shares in the vault instead of using the
+        // value returned by initializeBuffer;
+        uint256 totalShares = vault.getBufferTotalShares(wrapper);
+
+        vm.prank(lp);
+        (uint256 underlyingDeposited, uint256 wrappedDeposited) = router.addLiquidityToBuffer(wrapper, sharesToIssue);
+
+        // Measures if the underlying and wrapped deposited on addLiquidityToBuffer are worth the same number of shares
+        // than in initialized liquidity (or less).
```
Comment (EndymionJkb) [inline pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol:234]:
```suggestion
        // Ensure the underlying and wrapped value returned from `removeLiquidityFromBuffer` is equal to or less than
        // the amount received from initialization.
```

File: pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol [hunk 4]
```diff
@@ -197,20 +212,90 @@ abstract contract ERC4626WrapperBaseTest is Test {
...
+        uint256 wrappedToInitialize,
+        uint256 sharesToRemove
+    ) public {
+        underlyingToInitialize = bound(underlyingToInitialize, 1e6, underlyingToken.balanceOf(lp) / 10);
+        wrappedToInitialize = bound(wrappedToInitialize, 1e6, wrapper.balanceOf(lp) / 10);
+
+        vm.prank(lp);
+        uint256 lpShares = router.initializeBuffer(wrapper, underlyingToInitialize, wrappedToInitialize);
+        // Since the buffer burns part of the shares, we measure the total shares in the vault instead of using the
+        // value returned by initializeBuffer;
```
Comment (EndymionJkb) [inline pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol:258]:
```suggestion
        // value returned by initializeBuffer.
```

File: pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol [hunk 5]
```diff
@@ -197,20 +212,90 @@ abstract contract ERC4626WrapperBaseTest is Test {
...
+        uint256 sharesToIssue
+    ) public {
+        underlyingToInitialize = bound(underlyingToInitialize, 1e6, underlyingToken.balanceOf(lp) / 10);
+        wrappedToInitialize = bound(wrappedToInitialize, 1e6, wrapper.balanceOf(lp) / 10);
+        sharesToIssue = bound(sharesToIssue, 1e6, underlyingToken.balanceOf(lp) / 2);
+
+        vm.prank(lp);
+        router.initializeBuffer(wrapper, underlyingToInitialize, wrappedToInitialize);
+        // Since the buffer burns part of the shares, we measure the total shares in the vault instead of using the
+        // value returned by initializeBuffer;
```
Comment (EndymionJkb) [inline pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol:227]:
```suggestion
        // value returned by initializeBuffer.
```

**Review by EndymionJkb — APPROVED** [review]
Just some minor comments.

**Review by elshan-eth — APPROVED** [review]
LGTM!

**Review by jubeira — APPROVED** [review]
Added one more that fully roundtrips add / remove with different sizes. LGTM!

**Pull request merged**